src/Psl/Hash/Hmac/hash.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Hash\Hmac;
+
+use Psl;
+use Psl\Arr;
+
+use function hash_hmac;
+
+/**
+ * Generate a keyed hash value using the HMAC method.
+ *
+ * @throws Psl\Exception\InvariantViolationException If the given algorithm is unsupported.
+ *
+ * @psalm-pure
+ */
+function hash(string $data, string $algorithm, string $key): string
```

Comment:
The reason I flipped the arguments here is to "match" the signature of `Password\hash` ( even tho $algorithm has a default value there ):

consistent:
```php
$hash = Password\hash($password, $algorithm);
$hash = Hash\hash($some_data, $algorithm);
```

inconsistent:

```php
$hash = Password\hash($password, $algorithm);
$hash = Hash\hash($algorithm, $some_data);
```

Reply:
I'dd keep it consistent. Those are things that can be prefilled by the IDE as well...
What bothers me with those hashing functions is what algorithms are available.

Sure, one can look it up - but would it make sense to create shortcut functions per algorithm?
Isn't that list more or less fixed?

Reply:
not really, PHP std stopped doing that function-per-algo thing a while ago, every few months PHP will add support for a couple of algorithms, this can get a bit messy, there's `Hash\algorithmes` function, we can add documentation of the currently available algorithms there i guess.

Reply:
full list: https://www.php.net/manual/en/function.hash-algos.php

Reply:
ok thats a big list - makes sense not to include it